File: MONTH-01 FOLDER/DAY-04/comparison.py
# --------------------------------------------------------
# DAY 04: PYTHON COMPARISON OPERATORS
# --------------------------------------------------------

print("--- Python Comparison Operators Running ---")

user_score = 90
passing_score = 50

# 1. Standard Comparisons
is_passing = user_score >= passing_score  # Greater than or equal to
failed = user_score < passing_score       # Less than

print(f"Did the user pass? {is_passing}")  # True
print(f"Did the user fail? {failed}")      # False

# 2. Equality Testing
# Python doesn't have === because it strictly prevents comparing strings to numbers implicitly.
stored_pin = 1234
entered_pin = 1234

is_pin_correct = (stored_pin == entered_pin)
print(f"Is the password match successful? {is_pin_correct}") # True
print(f"Is the pin wrong? {stored_pin != entered_pin}")       # False

File: MONTH-01 FOLDER/DAY-04/comparison.js
// --------------------------------------------------------
// DAY 04: JAVASCRIPT COMPARISON OPERATORS
// --------------------------------------------------------

console.log("--- JS Comparison Operators Running ---");

let myAge = 20;
let legalAge = 18;
let stringAge = "20";

// 1. Basic Comparisons (Returns true or false)
console.log("Is my age greater than legal age?", myAge > legalAge); // true
console.log("Is my age less than legal age?", myAge < legalAge);    // false

// 2. The Golden JS Rule: == vs ===
// Loose equality (==) only checks the VALUE, ignoring the data type.
console.log("Loose Equality (20 == '20'):", myAge == stringAge);   // true 

// Strict equality (===) checks BOTH the VALUE and the DATA TYPE (Highly recommended!)
console.log("Strict Equality (20 === '20'):", myAge === stringAge); // false (Number vs String)

// 3. Not Equal (!= and !==)
console.log("Strict Not Equal (20 !== '20'):", myAge !== stringAge); // true
